authors/apps/authentication/backends.py

```diff
@@ -9,16 +9,61 @@
 """Configure JWT Here"""
 
 
-class JWTAuthentication:
+class JWTAuthentication(authentication.BaseAuthentication):
+    authentication_header_prefix = 'Token'
 
-    def __init__(self):
+    def authenticate(self, request):
         """
-        Create a class instance here
+        method called on every request regardless of
+        whether the endpoint requires authentication.
         """
-        pass
+        request.user = None
 
-    def authenticate(self, user_token):
+        auth_header = authentication.get_authorization_header(request).split()
+        auth_header_prefix = self.authentication_header_prefix.lower()
+
+        if not auth_header:
+            return None
+
+        if len(auth_header) == 1:
+            # Invalid token header. No credentials provided.
+            return None
+
+        elif len(auth_header) > 2:
+            # Invalid token header. The Token string should not contain spaces.
+            return None
+
+        # decode prefix and token values.
+        prefix = auth_header[0].decode('utf-8')
+        token = auth_header[1].decode('utf-8')
+
+        if prefix.lower() != auth_header_prefix:
+            # The auth header prefix is not what we expected
+            return None
+
+        return self._authenticate_credentials(request, token)
+
+    def _authenticate_credentials(self, request, token):
         """
-        Check Authenticity of a user token here
+        Method to try and authenticate the given credentials.
         """
-        pass
+        try:
+            payload = jwt.decode(token, settings.SECRET_KEY)
+        except:  # noqa: E722
+            msg = 'Invalid authentication. Could not decode token.'
+            raise exceptions.AuthenticationFailed(msg)
+
+        try:
+            user = User.objects.get(pk=payload['id'])
+        except User.DoesNotExist:
+            msg = 'No user matching this token was found.'
+            raise exceptions.AuthenticationFailed(msg)
+
+        if not user.is_active:
+            msg = 'This user has been deactivated.'
```

Comment:
I would recommend using a message like `msg = please, check your email to activate your account`. This is because a user should only be deactivated after he/she deletes his/her account and not when his/her account is not yet verified by email. In the modal.py file, the is a database row created named `is_active`. Check it out and read the documentation above that line.

Reply:
Yes, so if we're to change this message, I think we could change it to `this account has been deactivated`. because we are not providing a reactivation feature for now. Users, I think, will just have to create a new account once they deactivated/deleted their account.

Reply:
@lytes20, There is a difference between a deactivated account and the situation where on ehas not confirmed their email address. For the case in the subject, the user has not yet approved their email but `is_active` in models.py is by default equal to boolean `True` and yet `is_account_verfied` in your models.py file is by default set to boolean `False`.

Reply:
My bad, @lytes20. Sorry, I misunderstood that line of code. You are actually right on that.